### ISSUE TITLE
payoff of addon or refi needs to hit original store

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1682]
_Imported from trac ticket 1682.
Created by **rneville, sal**
Opened in trac: Jun 3, 2012, 3:19:39 PM
Last modified in trac: Jul 13, 2012, 9:32:02 AM_
***
_by rneville, sal:_
Going way back we had an issue when one store does a payoff for another store it was putting a payment store of the store that took the payoff payment on the interest and finance rebates. Those transactions actually go back to the original store that wrote the contract. I believe that was fixed but it is coming back now and it appears to be on add-on’s.
 
See contracts 0415143529 and 1001158970

**Comment by kbower** [comment]
_**rneville** trac comment on Jun 3, 2012, 3:19:59 PM:_

If I had to guess, it's probably putting the payment store of the addon contract, not the original contract. There's a snip of code that runs to set the default values for all ArTran records that has a spot like this:
```python
 if self.contractid:
            self.artype = ArTran.ARTYPE_INSTALLMENT
            if contract:
                self.sourcesiteid = contract.siteid
```
contract in this instance will be the immediate contract tied to the contractid in the PMT transaction (i.e. it will be the addon contract in the case you're talking about).

Should this bit instead be universally seeking out the base-original contract's siteid?

**Comment by kbower** [comment]
_**rneville** trac comment on Jun 3, 2012, 3:20:14 PM:_

I guess so, the problem is when it’s not actual money being handled and it’s just
 a rebate like this instance, putting the rebate on the wrong store causes accounting dept. issues.

**Comment by kbower** [comment]
_**rneville** trac comment on Jun 13, 2012, 8:55:00 AM:_

Ticket [#1712](1712) has been marked as a duplicate of this ticket.

**Comment by kbower** [comment]
_**rneville** trac comment on Jul 13, 2012, 9:32:02 AM:_

(Changeset [[2867]](../commit/1307281fba1cc1576c36d1c1ea331843f87eda49))
 * fixes #1682 (payoff of addon or refi needs to hit original store)

**Comment by kbower** [comment]
_**rneville** trac status change on Jul 13, 2012, 9:32:02 AM:_ **closed**
